engine/src/main/scala/cromwell/services/ServiceRegistryActor.scala

```diff
@@ -0,0 +1,46 @@
+package cromwell.services
+
+import akka.actor.{Actor, ActorRef, Props}
+import com.typesafe.config.{ConfigFactory, Config, ConfigObject}
+import cromwell.engine.db.DataAccess
+import cromwell.services.ServiceRegistryActor._
+import lenthall.config.ScalaConfig._
+
+import scala.collection.JavaConverters._
+
+object ServiceRegistryActor {
+  case class Send(serviceName: String, message: Any)
+  case class SendFailure(serviceName: String)
+
+  def props(dataAccess: DataAccess, config: Config) = {
+    Props(ServiceRegistryActor(dataAccess, config))
+  }
+}
+
+case class ServiceRegistryActor(dataAccess: DataAccess, globalConfig: Config) extends Actor {
+  val services: Map[String, ActorRef] = {
+    val serviceNamesToConfigStanzas = globalConfig.getObject("services").entrySet.asScala.map(x => x.getKey -> x.getValue).toMap
+    serviceNamesToConfigStanzas map {
+      case (serviceName, config: ConfigObject) => serviceName -> makeActor(serviceName, config.toConfig)
+      case (serviceName, _) => throw new Exception(s"invalid configuration for service $serviceName")
+    }
+  }
+
+  private def makeActor(serviceName: String, serviceStanza: Config): ActorRef = {
+    val serviceConfigStanza = serviceStanza.getConfigOption("config").getOrElse(ConfigFactory.parseString(""))
+    val className = serviceStanza.getStringOption("class").getOrElse(
+      throw new Exception(s"invalid configuration for service $serviceName: missing 'class' definition")
+    )
+    val method = Class.forName(className).getMethod("props", classOf[DataAccess], classOf[Config], classOf[Config])
+    val props = method.invoke(None, dataAccess, serviceConfigStanza, globalConfig).asInstanceOf[Props]
+    context.actorOf(props)
+  }
+
+  def receive = {
+    case Send(serviceName, message) =>
```

Comment:
Just a thought which might be horrible... It might be possible to remove this awkward "Send" message by including the serviceName in the message. You could have:
```
trait ServiceRegistryMessage { def serviceName: String }
```

And then every service's messages could just have (eg):
```
sealed trait KvMessage extends ServiceRegistryMessage { val serviceName = "KeyValue" }
```